oauth2_client/models.py

```diff
@@ -1,54 +1,251 @@
+    Application model used by the client
     """
-    name = models.CharField(max_length=50, help_text="Example: licence")
-    # base definitions copied from oauth2_provider.Application
-    client_id = models.CharField(max_length=100)
-    client_secret = models.CharField(max_length=100)
+    GRANT_CLIENT_CREDENTIALS = "client-credentials"
+    GRANT_JWT_BEARER = "jwt-bearer"
+    GRANT_TYPES = (
+        (GRANT_CLIENT_CREDENTIALS, _("Client credentials")),
+        (GRANT_JWT_BEARER, _("JWT bearer")),
+    )
+    name = models.CharField(
+        max_length=255,
+        unique=True,
+        help_text="Application name"
+    )
+    client_id = models.CharField(
+        max_length=100,
+        help_text="ID string of the application"
+    )
+    client_secret = models.CharField(
+        max_length=255,
+        help_text="Secret string of this application"
+    )
+    authorization_grant_type = models.CharField(
+        max_length=32,
+        choices=GRANT_TYPES,
+        help_text="The type of authorization grant to be used"
+    )
     # URLField isn't used because it doesn't accept docker-to-docker domains
     service_host = models.CharField(
         max_length=200,
-        help_text="Service host. For example: http://service-a:8000 or https://serive-a.labster.com")
-    # URLField isn't used because it doesn't accept docker-to-docker domains
-    token_url = models.CharField(
+        help_text="Service host. For example: http://service-a:8000 or https://serive-a.labster.com"
+    )
+    token_uri = models.CharField(
         max_length=200,
         help_text="Token URL. For example: http://service-a:8000/o/token/"
     )
-
+    # Scope is a series of space delimited strings, eg. "read write"
     scope = models.CharField(
         max_length=100,
         blank=True,
         default="",
-        help_text="Custom permissions string. Empty value is allowed, and set by default."
+        help_text="Scope to be requested from provider, as a series of space delimited strings, "
+                  "e.g. `read write`"
     )
-
-    updated_at = models.DateTimeField(auto_now=timezone.now)
-    created_at = models.DateTimeField(auto_now_add=timezone.now)
+    extra_settings = JSONField(
+        default=dict,
+        blank=True,
+        help_text="Custom settings JSON string. For data that didn't fit anywhere else"
+    )
+    created = models.DateTimeField(auto_now_add=True)
+    updated = models.DateTimeField(auto_now=True)
 
     def __unicode__(self):
         """
         Returns the application name to show in the admin.
         """
         return self.name
 
+    def __str__(self):
+        return "<Application, name:%s>" % self.name
+
+    def subject_or_none(self):
+        """
+        Get subject from extra_settings JSON field.
+
+        Returns:
+            str: subject or None
+        """
+        return self.extra_settings['subject'] if 'subject' in self.extra_settings else None
```

Comment:
if 'subject' in self.extra_settings:
        return self.extra_settings['subject']

 Better:

    return self.extra_settyings.get('subject')